google/cloud/security/inventory/inventory_loader.py

```diff
@@ -372,45 +242,68 @@ def _send_email(cycle_time, cycle_timestamp, status, pipelines,
         LOGGER.error('Unable to send email that inventory snapshot completed.')
 
 
-def _configure_logging(configs):
+def _configure_logging(flag_configs):
     """Configures the loglevel for all loggers."""
-    desc = configs.get('loglevel')
+    desc = flag_configs.get('loglevel')
     level = LOGLEVELS.setdefault(desc, 'info')
     log_util.set_logger_level(level)
 
 def main(_):
     """Runs the Inventory Loader."""
-    try:
-        dao = Dao()
-        project_dao = proj_dao.ProjectDao()
-        organization_dao = org_dao.OrganizationDao()
-        bucket_dao = buck_dao.BucketDao()
-        cloudsql_dao = sql_dao.CloudsqlDao()
-        fwd_rules_dao = fr_dao.ForwardingRulesDao()
-        folder_dao = folder_resource_dao.FolderDao()
-    except data_access_errors.MySQLError as e:
-        LOGGER.error('Encountered error with Cloud SQL. Abort.\n%s', e)
+
+    flag_configs = FLAGS.FlagValuesDict()
+    _configure_logging(flag_configs)
+
+    if FLAGS.pipeline_configs is None:
+        LOGGER.error('Path to pipeline config needs to be specified.')
         sys.exit()
 
-    cycle_time, cycle_timestamp = _start_snapshot_cycle(dao)
+    # Make these re-usable so that the rate-limiter can apply across
+    # different pipelines.
+    try:
+        api_map = {
+            'admin_api': admin_directory.AdminDirectoryClient(),
+            'bigquery_api': bigquery.BigQueryClient(),
+            'cloudsql_api': cloudsql.CloudsqlClient(),
+            'compute_api': compute.ComputeClient(),
+            'compute_beta_api': compute.ComputeClient(version='beta'),
+            'crm_api': crm.CloudResourceManagerClient(),
+            'crm_v2beta1': crm.CloudResourceManagerClient(version='v2beta1'),
+            'gcs_api': gcs.StorageClient(),
+        }
+    # A lot of potential errors can be thrown by different APIs.
+    # So, handle generically.
+    except:
+        LOGGER.error('Error to create the api map.\n%s', sys.exc_info()[0])
+        sys.exit()
 
-    configs = FLAGS.FlagValuesDict()
+    # Make these re-usable so that the db connection can apply across
+    # different pipelines.
+    try:
+        dao_map = {
+            'bucket_dao': bucket_dao.BucketDao(),
+            'cloudsql_dao': cloudsql_dao.CloudsqlDao(),
+            'dao': dao.Dao(),
+            'folder_dao': folder_dao.FolderDao(),
+            'forwarding_rules_dao': forwarding_rules_dao.ForwardingRulesDao(),
+            'organization_dao': organization_dao.OrganizationDao(),
+            'project_dao': project_dao.ProjectDao(),
+        }
+    except data_access_errors.MySQLError as e:
+        LOGGER.error('Encountered error to create dao map.\n%s', e)
+        sys.exit()
 
-    _configure_logging(configs)
+    cycle_time, cycle_timestamp = _start_snapshot_cycle(dao_map.get('dao'))
```

Comment:
nit: you fetch 'dao' a few times from dao_map below, want to store it in a variable?